rag/advanced_rag/harness/sufficiency.py

```
@@ -25,79 +25,297 @@ def extract_numbers(text: str) -> list[float]:
+    can fall back to the regex path without crashing.
+    """
+    model_name = _resolve_spacy_model(language)
+    if model_name in _spacy_nlp_cache:
+        nlp = _spacy_nlp_cache[model_name]
+    else:
+        try:
+            import spacy
+
+            nlp = spacy.load(model_name)
+            _spacy_nlp_cache[model_name] = nlp
+        except Exception as exc:  # model not installed / spacy unavailable
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)